### ISSUE TITLE
DialogueRunner's OnNodeComplete() not running 

### ISSUE THREAD

**Issue by cory-devs** [YarnSpinnerTool/YarnSpinner-Unity#82]
Hey YarnSpinner team, 

I'm rather new to this so it's possible this is more of a feature, not a bug, but I'm having an issue in my YarnSpinner script where OnNodeComplete() isn't running for a function attached to the Dialogue Runner. 

My script looks like this when it doesn't run: 

```
title: Vanilla
tags:
---
<<if visited("Vanilla") is false>>
    Nightshade: Hey, Vanilla.
    Vanilla: Glad to see you back. 
    Nightshade: Anything I can do to help? Things feel out of sorts.
    Vanilla: Of course! If you could get us some more wood it'd be appreciated.
    Vanilla: About 20 pieces should do. 
<<else>>
    Vanilla: Hey!
<<endif>>

<<if visited("Vanilla") == true && inventory_count("Wood", 20) == false>>
    Vanilla: Still waiting on that wood. 
    Vanilla: Remember: 20 pieces should be enough.
    Vanilla: Thanks for the help! 
<<endif>>

<<if visited("Vanilla") == true && inventory_count("Wood", 20) == true>>
    Vanilla: You got it. 
    Vanilla: I appreciate it a lot.
    [[Anything else I can do?|AnythingElse]]
    [[Glad I could help!|Vanilla.GladHelp]]
<<endif>>
===

title: AnythingElse
tags: 
---
Vanilla: Not right now. I'll let you know. Thanks again!
===

title: Vanilla.GladHelp
tags: 
---
Vanilla: I'll let you know if I need anything else! Thanks again!
===
```

But if I remove the second two if statements from the "Vanilla" node OnNodeComplete() runs fine. 

I'm using YarnSpinner 1.2.7 and Unity 2019.4.3f1 Personal. 

Let me know if there is any more information you need, or formatting you'd like me to try. Even if this is how it's intended to work I'm not getting an error or anything telling me something is wrong and that by itself would be a useful feature.

EDIT: I made sure to add a jump to another node before the end of the Vanilla node and now I get the OnNodeComplete() running fine. But, if I remove that, the problem persists. I'm glad I seemed to find the solution but it seems like OnNodeComplete() should run no matter what at the end of the node?

**Comment by desplesda** [comment]
I'm not seeing this behaviour in Yarn Spinner 2.0 beta 4; if possible, I'd suggest upgrading. I'll close this issue here, but if you're still encountering problems in 2.0, please file a new issue! Thanks for filing the ticket!
